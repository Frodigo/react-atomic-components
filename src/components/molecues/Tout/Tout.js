import React from 'react';
import PropTypes from 'prop-types';

import Heading from '../../atoms/Text/Heading';

import classes from './Tout.module.css';

const Tout = (props) => {
    const {
        imageUrl,
        imageWidth,
        imageHeight,
        linkUrl,
        title,
        displayTape,
        ...rest
    } = props;

    const shouldDisplayTape = displayTape ? (
        <header className={classes.tape}>
            <Heading type="h5" classes="inverse">
                {title}
            </Heading>
        </header>
    ) : null;

    return (
        <div className={classes.tout} {...rest}>
            <a href={linkUrl}>
                <figure className={classes.figure}>
                    <img
                        src={imageUrl}
                        width={imageWidth}
                        height={imageHeight}
                        layout="responsive"
                        alt="figure"
                    />
                </figure>
                {shouldDisplayTape}
            </a>
        </div>
    );
};

Tout.propTypes = {
    displayTape: PropTypes.bool,
    imageUrl: PropTypes.string.isRequired,
    linkUrl: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

Tout.defaultProps = {
    displayTape: true
};

export default Tout;
