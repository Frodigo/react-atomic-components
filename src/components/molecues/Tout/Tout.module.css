.tout {
    min-height: 270px;
    position: relative;
}

.figure {
    margin: 0;
    padding: 0;
}

.figure img {
    display: block;
    width: 100%;
}

.tape {
    background-color: var(--color-secondary);
    bottom: 0;
    left: 0;
    padding: var(--spacing-md);
    position: absolute;
    right: 0;
}

@media (min-width: 768px) {
    .tout {
        min-height: 340px;
    }
}

@media (min-width: 1024px) {
    .tout {
        min-height: 400px;
    }
}

@media (min-width: 1300px) {
    .tape {
        transition: .3s background-color;
    }

    .tout:hover .tape,
    .tout:focus .tape {
        background-color: var(--color-primary);
    }
}