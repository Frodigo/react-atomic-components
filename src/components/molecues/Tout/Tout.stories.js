import React from 'react';

import { default as ToutComponent } from './Tout';

export default {
    title: 'Molecues/Tout',
    component: ToutComponent
};

const Template = (args) => <ToutComponent {...args} />;

export const Tout = Template.bind({});
Tout.args = {
    imageUrl: 'https://via.placeholder.com/600x400',
    linkUrl: '#',
    title: 'Lorem ipsum dolor sit',
    imageWidth: 600,
    imageHeight: 400
};
