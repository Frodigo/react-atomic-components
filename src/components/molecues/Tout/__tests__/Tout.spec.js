import React from 'react';
import { render } from '@testing-library/react';

import Tout from '../Tout';

describe('Tout should be rendered', () => {
    const props = {
        imageUrl: 'https://via.placeholder.com/290x240',
        linkUrl: '#',
        title: 'Lorem ipsum dolor sit'
    };

    test('Tout renders correctly', () => {
        const { container } = render(<Tout {...props} />);

        expect(container).toMatchSnapshot();
    });
});
