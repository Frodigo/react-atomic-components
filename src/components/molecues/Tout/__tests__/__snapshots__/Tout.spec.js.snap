// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tout should be rendered Tout renders correctly 1`] = `
<div>
  <div
    class="tout"
  >
    <a
      href="#"
    >
      <figure
        class="figure"
      >
        <img
          alt="figure"
          layout="responsive"
          src="https://via.placeholder.com/290x240"
        />
      </figure>
      <header
        class="tape"
      >
        <h5
          class="inverse"
        >
          Lorem ipsum dolor sit
        </h5>
      </header>
    </a>
  </div>
</div>
`;
